gui/default/syncthing/core/localeService.js

```diff
@@ -78,11 +78,16 @@ angular.module('syncthing.core')
                                 // case. We compare to the lowercase version of the language
                                 // code we have as well.
                                 possibleLang = possibleLang.toLowerCase();
-                                if (possibleLang.length > lang.length) {
-                                    return possibleLang.indexOf(lang) === 0;
-                                } else {
-                                    return lang.indexOf(possibleLang) === 0;
+                                if (possibleLang.indexOf(browserLang) !== 0) {
+                                    // Prefix does not match.
```

Comment:
```suggestion
                                    // Prefix does not match
```
Please let's not get started on this.  It's just code comments and for such short notes, we really don't need to pretend it's part of a prose text.  Please stick to the established style in this point.

Reply:
All fixed, although to be honest, there is no established style (especially if you go through all the JS files, not just this one). I think it all depends on who at what time wrote the specific code and comments.

It would be nice to have a specific style for HTML and JS files defined in the Docs though, which everyone would be expected to follow, as currently it really is all over the place…

Reply:
Yes, something similar to `gofmt` would be helpful.  But before we start putting effort into such a thing, we should try to modernize and get to a more future-proof framework basis.  AngularJS in the currently used version is pretty outdated and our use of it rather monolithic, where individual components would help to better structure it.

Reply:
Yeah, but moving the GUI to another framework would basically mean a fulltime job, so I highly doubt something like that will happen in the near future, if ever. It may be easier to rewrite the whole thing from scratch 😅 (which will also likely require a lot of time).

I mean very basic things like 1) Don't use full stops in single-line comments, 2) Limit comments to xx characters per line, 3) Indent the code with x spaces / tabs, 4) Use HTML5 only, etc. Basically something like https://google.github.io/styleguide/jsguide.html#formatting. Maybe simply choose an established style guide for each language and link to it?

One advantage of the current Web GUI code is that it's been tested thoroughly in various environments. It's old-school but at same time stable, and the core doesn't change at all, so you know what to expect.

Reply:
> Anyway, this part of the discussion probably belongs to the forum, not here 🙂.

Absolutely :-)

I've quickly read over the referenced Google Style guidelines.  They sound sensible AFAICS.  Very elaborate, as in many things to remember if you're just an occasional JS programmer.  But specifically the topic of the full stop in (inline code) comments is not clarified, only for JSDoc strings.  I stand by my opinion that it's just superfluous because these inline comments (in contrast to API docs) are never looked at outside of the code's context.

I would welcome an official recommendation on which style guide to follow, but that should be discussed on the forum, and also where to document it.